# Momentum experiment 26-Nov-2014 (a)

This workbbok is a follow-up to http://pdf.markcarter.me.uk/momo2014-11-26.html. The data is the same, but we try to answer this question: **what is the expected return on investing in each decile?**

In [1]:
%matplotlib inline
import matplotlib.mlab as mlab
import matplotlib.pyplot as plt
import pylab
import numpy as np
import pandas as pd

Load the data, which was obtained on 26-Nov-2014, or thereabouts:

In [2]:
df = pd.io.parsers.read_csv("http://pastebin.com/raw.php?i=JWBf8LzF")

Calculate the relative momentum for the the first sixth months:

In [3]:
df['rs6ma'] = (df.rs1y/100.0 + 1.0)/(df.rs6mb/100.0 + 1.0)*100.0 - 100.0
df.head()

epic       mkt  rs6mb   rs1y   rs5y      rs6ma
0   3IN   1322.84  10.00  14.31  17.29   3.918182
1   888    484.86   9.01 -15.17   0.03 -22.181451
2   AA.   1941.45    NaN    NaN    NaN        NaN
3   AAL  18564.58 -14.44  -3.60 -60.58  12.669472
4  ABBY    185.13  -1.87   0.73  67.17   2.649547

Create a function which, given a dataframe `df`, and the number of buckets `n` (e.g. `n=10` for deciles), returns another dataframe of the means of those buckets. This is the expected return on each quantile:

In [4]:
def agg(df, n):
    df['tile'] = pd.qcut(df['rs6ma'], n, labels = range(n))
    res = df.groupby('tile')['rs6mb'].mean()
    return res

Let's explore the expected returns when we split the data into a variety of n-tiles:

In [14]:
for i in [2,3, 4, 5, 10, 20]:
    print(agg(df,i))
    print()

tile
0   -1.927321
1    1.822181
Name: rs6mb, dtype: float64

tile
0   -3.821262
1   -0.215187
2    3.878738
Name: rs6mb, dtype: float64

tile
0   -4.854534
1    1.018187
2   -0.932250
3    4.559503
Name: rs6mb, dtype: float64

tile
0   -5.974574
1    0.638828
2   -1.016953
3    0.137656
4    5.951550
Name: rs6mb, dtype: float64

tile
0    -8.968615
1    -2.933750
2    -0.834063
3     2.111719
4     1.098125
5    -3.132031
6    -0.999375
7     1.274688
8     1.648125
9    10.188769
Name: rs6mb, dtype: float64

tile
0    -16.913939
1     -0.775000
2      0.921875
3     -6.789375
4     -0.339375
5     -1.328750
6     -0.773438
7      4.996875
8      3.688125
9     -1.491875
10    -3.897500
11    -2.366563
12    -3.061250
13     1.062500
14     3.601563
15    -1.052188
16     5.445000
17    -2.148750
18    17.644375
19     2.959091
Name: rs6mb, dtype: float64



## Conclusion

As you can see, if you split the data into two, you obtain a slight outperformance if you invest in the highest previous performers: 1.8% as opposed to -1.9% over the next six months.

As the number of quantiles increases, it becomes apparent that the best and worst performers lead to the best and worst subsequent performances. The intermediate groups are mixed, and do not show much of a pattern.

## About this document

    Author:  Mark Carter
    Created: 25-Oct-2015